cloud/pkg/edgecontroller/controller/downstream.go

```diff
@@ -545,6 +548,61 @@ func (dc *DownstreamController) syncRuleEndpoint() {
 	}
 }
 
+func (dc *DownstreamController) syncMission() {
+	var operation string
+	for {
+		select {
+		case <-beehiveContext.Done():
+			klog.Warning("Stop edgecontroller downstream syncMission loop")
+			return
+		case e := <-dc.missionsManager.Events():			
+			klog.V(4).Infof("Get mission events: event type: %s.", e.Type)
+			mission, ok := e.Object.(*missionsv1.Mission)
+			if !ok {
+				klog.Warningf("object type: %T unsupported", mission)
+				continue
+			}
+			klog.V(4).Infof("Get mission events: mission object: %+v.", mission)
+			switch e.Type {
+			case watch.Added:
+				operation = model.InsertOperation
+			case watch.Modified:
+				operation = model.UpdateOperation
+			case watch.Deleted:
+				operation = model.DeleteOperation
+			default:
+				// unsupported operation, no need to send to any node
+				klog.Warningf("Mission event type: %s unsupported", e.Type)
+				continue
+			}
+
+			// send to all nodes
+			dc.lc.EdgeNodes.Range(func(key interface{}, value interface{}) bool {
+				nodeName, ok := key.(string)
+				if !ok {
+					klog.Warning("Failed to assert key to sting")
+					return true
+				}
+				msg := model.NewMessage("")
+				msg.SetResourceVersion(mission.ResourceVersion)
+				resource, err := messagelayer.BuildResource(nodeName, "fake_namespace", model.ResourceTypeMission, mission.Name)
+				if err != nil {
+					klog.Warningf("Built message resource failed with error: %v", err)
+					return true
+				}
+				msg.BuildRouter(modules.EdgeControllerModuleName, constants.GroupResource, resource, operation)
+				msg.Content = mission
+				if err := dc.messageLayer.Send(*msg); err != nil {
+					klog.Warningf("Send message failed with error: %s, operation: %s, resource: %s", err, msg.GetOperation(), msg.GetResource())
+				} else {
+					klog.V(4).Infof("Send message successfully, operation: %s, resource: %s", msg.GetOperation(), msg.GetResource())
```

Comment:
"message sent successfully"

also I'm a bit confused about this whole section. what does "all nodes" mean here?

Reply:
Updated all the "Send message successfully" in the code. 

"all the nodes" means all the edge nodes in the cluster. One edge controller may talk to multiple edge nodes.